packages/babel-parser/src/plugins/typescript/index.ts

```diff
@@ -3393,7 +3403,31 @@ export default (superClass: ClassWithMixin<typeof Parser, IJSXParserMixin>) =>
         }
       }
 
-      return super.parseConditional(expr, startLoc, refExpressionErrors);
+      this.next(); // eat '?'
+
+      const node = this.startNodeAt<N.ConditionalExpression>(startLoc);
+      node.test = expr;
+
+      const state = this.state.clone();
+      this.state.noArrowAt = [];
+
+      node.consequent = this.parseMaybeAssignAllowIn();
+
+      if (!this.eat(tt.colon)) {
+        state.noArrowAt = this.state.noArrowAt;
+        if (state.noArrowAt.length <= 3) {
+          state.noArrowAt = [state.noArrowAt[0]];
+        } else {
+          state.noArrowAt = state.noArrowAt.slice(3);
+        }
+        this.state = state;
+        node.consequent = this.parseMaybeAssignAllowIn();
+        this.expect(tt.colon);
+      }
```

Comment:
This was inspired by the Flow plugin, thanks to @nicolo-ribaudo.
The current implementation doesn't look reliable at all, but it passes all the tests.🤦‍♂️
If we try them one by one, it will be more reliable, but slower.